src/zino/statemodels.py

```diff
@@ -80,6 +80,13 @@ def is_cisco(self):
     def is_juniper(self):
         return self.enterprise_id == 2636
 
+    def set_boot_time_from_uptime(self, uptime: int):
+        """Calculates and sets the device boot time from a current uptime value.
+
+        :param uptime: An uptime value in 100ths of a second
+        """
+        self.boot_time = datetime.datetime.now() - datetime.timedelta(seconds=uptime / 100)
```

Comment:
Actually, I would change this slightly, now that I see it again in hindsight.

We already have a utility function to ensure we generate timestamps with timezone info: `zino.time.now()`